Gems/Atom/Tools/AtomToolsFramework/Code/Source/Window/AtomToolsMainWindow.cpp

```diff
@@ -195,22 +195,22 @@ namespace AtomToolsFramework
         if (!m_updateMenus)
         {
             m_updateMenus = true;
-            QTimer::singleShot(0, [this]() {
+            QTimer::singleShot(0, this, [this]() {
```

Comment:
Why leave this one as a `QTimer::singleShot` rather than put it on the SystemTickBus?

Reply:
This one still requires access to members like the menu bar. By adding this as the second parameter, it's supposed to guarantee that the timer will be cancelled if this is destroyed before it executes. I will continue to look at rearranging it so that I can get rid of the timer completely. The main reason to switch away from timers for some things is the guarantee synchronous updates In sync with the system tick. But that also requires that the lambda or callback function is not holding on to pointers or any other data that might be invalidated before the tick or timer triggers. That's why I changed some things to use buses that won't have any effect if the handler is destroyed or disconnected.